### ISSUE TITLE
Program edit screen narrator while typing support

### ISSUE THREAD

**Issue by TeamDman** [TeamDman/SuperFactoryManager#254]
The narrator experience in the editor sucks.
Tab navigation in the editor is broken to make tab-indent work.
Usual narration tools like NVDA and Windows narrator will say characters and stuff as you type.

When the cursor moves, the character to the right of the caret should be spoken.
When the selection changes, the newly selected or unselected content should be spoken.
